django/core/servers/basehttp.py

```diff
@@ -129,7 +130,8 @@ def __init__(self, stdin, stdout, stderr, environ, **kwargs):
         )
 
     def cleanup_headers(self):
-        super().cleanup_headers()
+        if self.environ["REQUEST_METHOD"] != "HEAD":
+            super().cleanup_headers()
```

Comment:
I'd move it to the `set_content_length()` method which is called by `simple_server.ServerHandler.cleanup_headers()`:
```python
def set_content_length(self):
    if self.environ["REQUEST_METHOD"] != "HEAD":
        super().set_content_length()
```

Reply:
... or maybe to always remove `Content-Length` (even if it's set in a different way):
```python
    def cleanup_headers(self):
        super().cleanup_headers()
        if self.environ["REQUEST_METHOD"] == "HEAD" and "Content-Length" in self.headers:
            del self.headers["Content-Length"]
        ...
```

Reply:
Good idea :+1: